index.js

```diff
@@ -52,79 +52,73 @@ const root =
   // clicked on (in terminals such as iTerm) to open the file in an editor.
   `${process.cwd()}/`
 
-expandThunks = (array) => array.map(
-  elem => typeof elem == 'function' ? elem() : elem
-)
+const PRETTY = Boolean(process.env.PRETTY)
 
-makeLazyLogger = (strictLogger) => {
-  return strictLogger.enabled ?
-    // if logger enabled, call it with functions expanded to actual values
-    (...args) => strictLogger.apply(strictLogger, expandThunks(args)) :
-    // no-op
-    () => {}
-}
+const DEBUG = process.env.DEBUG
 
-makeStrictLogger = (label, {logToStderr, isErrorLogger}) => {
-  const logger = debug(label)
-  if (!logToStderr && !process.env.DEBUG_LOG_TO_STDERR) {
-    // by default log to stdout
-    logger.log = console.log.bind(console);
-  }
+const LEVELS = ['DEBUG', 'LOG', 'WARN', 'ERROR', 'EMERGENCY']
 
-  if (isErrorLogger) {
-    const errorLogger = (...args) => {
-      if (!logger.enabled) return
-
-      const last = args[ args.length - 1 ]
-
-      if (last instanceof Error) {
-        const argsWithErrorMessage = args.slice(0, -1)
-        argsWithErrorMessage.push(last.toString())
-        // Log error.message via the logger with label.
-        logger.apply(logger, argsWithErrorMessage)
-        // Log error object on its own, so that it can be parsed by GCP Error
-        // reporting.
-        console.error(last)
-      }
-      else {
-        logger(args)
-      }
-    }
+const expandThunks = array => array.map(elem => (typeof elem == 'function' ? elem() : elem))
 
-    errorLogger.enabled = logger.enabled
-    return errorLogger
-  }
+const makeLazyLogger = strictLogger => {
+  // call it with functions expanded to actual values
+  return (...args) => strictLogger.apply(strictLogger, expandThunks(args))
+}
 
+const parseArgs = args =>
+  args
+    .map(arg =>
+      stringify(
+        arg instanceof Error
+          ? {
+              name: arg.name,
+              message: arg.message,
+              data: arg.data,
+              stack: arg.stack
+            }
+          : arg
+      )
+    )
+    .join(' ')
+
+const makeStrictLogger = (severity, context) => {
+  const logger = (...args) => {
+    const message = parseArgs(args)
+    const payload = { severity, timestamp: Date.now(), context, message }
+
+    if (
+      (!DEBUG && severity !== 'DEBUG') ||
+      (DEBUG && !LEVELS.includes(DEBUG) && severity !== 'DEBUG') ||
+      DEBUG === '*' ||
+      (LEVELS.includes(DEBUG) && DEBUG === severity)
+    ) {
+      const log_args = [payload]
+      if (PRETTY) log_args.push(null, 2)
```

Comment:
Can we use a different logger format in case of `PRETTY`? 
```
timestamp | context: message | JSON.stringify(args)
```
instead of a json dump of everything to make it much more readable locally since it won't be auto parsed